samples/bluetooth/peripheral_hr/src/main.c

```diff
@@ -152,6 +207,25 @@ int main(void)
 
 		/* Battery level simulation */
 		bas_notify();
+
+		if (atomic_test_and_clear_bit(state, STATE_CONNECTED)) {
+			/* Connected callback executed */
+
+		} else if (atomic_test_and_clear_bit(state,
+						     STATE_DISCONNECTED)) {
```

Comment:
This looks weird. 

When we are connected, state is set to `STATE_CONNECTED`, but we then clearing it here? So after one iteration, we are still connected, but `STATE_CONNECTED` is not set.